napari/layers/base/base.py

```diff
@@ -230,6 +230,11 @@ def __init__(
         if name is None and data is not None:
             name = magic_name(data)
 
+        if scale and 0 in scale:
+            raise Exception(
+                f'Layer "{name}" is invalid because it has scale values of 0. The layer\'s scale is currently "{scale}"'
```

Comment:
```suggestion
                f"Layer {name!r} is invalid because it has scale values of 0. The layer's scale is currently {scale!r}"
```

(totally optional side note, if you find yourself escaping quotes, because you're already using both `'` and `"`, you can use the `!r` format string (meaning `repr()`) ... which will add quotes around strings)